traits/trait_handlers.py

```diff
@@ -920,323 +914,3 @@ def get_editor(self, trait):
             editors.append(editor)
 
         return CompoundEditor(editors=editors)
-
-    def items_event(self):
-        return items_event()
```

Comment:
Same comment as above - i'm not sure why this (and the `items_event` public function) have been removed.

Reply:
> I can take a closer look.

So: prior to this PR, both `TraitCompound` and `TraitList` make use of `items_event`. We're getting rid of `TraitList`, so that use no longer exists. `TraitCompound` should only ever be using this in the case that one of the items in the compound trait is a `TraitList` (there are some `TraitList` tests that exercise this case), and given that `TraitList` is being removed, that again shouldn't be a possibility.

A new-style `List` (or `Dict` or `Set`) inside a `TraitCompound` would likely go badly, but I don't think that's ever been a supported case.

Nothing outside of `trait_handlers` imports `items_event` (though there are other functions called `items_event` elsewhere).

So I think removing it is safe. I also don't see good alternatives. We could leave a stub function that doesn't use `TraitList`, but then what trait would it return?

Reply:
That makes sense now - it didn't occur to me earlier that the `items_event` method and `has_items` update in `TraitCompound` were there to support a `TraitList` within. Resolving.